modules/consensus/validtransaction.go

```diff
@@ -174,42 +175,70 @@ func (s *State) validSiafunds(t types.Transaction) (err error) {
 	return
 }
 
+// ValidStorageProofs checks that the storage proofs are valid in the context
+// of the consensus set.
+func (s *State) ValidStorageProofs(t types.Transaction) (err error) {
+	id := s.mu.RLock()
+	defer s.mu.RUnlock(id)
+	return s.validStorageProofs(t)
+}
+
 // validTransaction checks that all fields are valid within the current
 // consensus state. If not an error is returned.
-func (s *State) validTransaction(t types.Transaction) (err error) {
+func (s *State) validTransaction(t types.Transaction) error {
+	// Skip transaction verification if the State is accepting trusted blocks.
+	if !s.fullVerification {
+		return nil
+	}
+
 	// StandaloneValid will check things like signatures and properties that
 	// should be inherent to the transaction. (storage proof rules, etc.)
-	err = t.StandaloneValid(s.height())
+	err := t.StandaloneValid(s.height())
 	if err != nil {
-		return
+		return err
 	}
 
 	// Check that each portion of the transaction is legal given the current
 	// consensus set.
 	err = s.validSiacoins(t)
 	if err != nil {
-		return
+		return err
 	}
 	err = s.validFileContractRevisions(t)
 	if err != nil {
-		return
+		return err
 	}
 	err = s.validStorageProofs(t)
 	if err != nil {
-		return
+		return err
 	}
 	err = s.validSiafunds(t)
 	if err != nil {
-		return
+		return err
 	}
 
-	return
+	return nil
 }
 
-// ValidStorageProofs checks that the storage proofs are valid in the context
-// of the consensus set.
-func (s *State) ValidStorageProofs(t types.Transaction) (err error) {
-	id := s.mu.RLock()
-	defer s.mu.RUnlock(id)
-	return s.validStorageProofs(t)
+// TryTransactions applies the input transactions to the consensus set to
+// determine if they are valid. An error is returned IFF they are not a valid
+// set in the current consensus set. The size of the transactions and the set
+// is not checked.
+func (s *State) TryTransactions(txns []types.Transaction) error {
```

Comment:
It's for the tpool to use, I thought it made sense to design the function here and put it to use when I tackle the tpool.